File: javascript/gemini/p12_nc.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const id_sequence = [10, 20, 30, 40, 50];

function getValueFromArray(array, index) {
    if (index < 0 || index >= array.length) {
        return -1;
    }
    return array[index];
}

rl.on('line', (line) => {
    const index = parseInt(line.trim(), 10);
    
    if (isNaN(index)) {
        process.exit(-1);
    }

    const result = getValueFromArray(id_sequence, index);
    process.exit(result);
});

File: python/gemini/p12_nc.py
import sys

def get_value_from_array(array, index):
    if index < 0 or index >= len(array):
        return -1
    return array[index]

if __name__ == "__main__":
    id_sequence = [10, 20, 30, 40, 50]
    
    try:
        line = sys.stdin.readline()
        if not line:
            sys.exit(-1)
            
        index = int(line.strip())
        result = get_value_from_array(id_sequence, index)
        sys.exit(result)
    except ValueError:
        sys.exit(-1)
